src/terminator/mod.rs

```diff
@@ -517,6 +594,38 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 self.write_primval(dest, PrimVal::Ptr(new_ptr), dest_ty)?;
             }
 
+            "__rust_maybe_catch_panic" => {
+                // We abort on panic, so not much is going on here, but we still have to call the closure
+                let u8_ptr_ty = self.tcx.mk_mut_ptr(self.tcx.types.u8);
+                let f = args[0].read_ptr(&self.memory)?;
+                let data = args[1].read_ptr(&self.memory)?; // FIXME: Why does value_to_primval(args[2], u8_ptr_ty)?.to_ptr()? here end up doing the Wrong Thing (TM)?
```

Comment:
Canyou add the signature of the function as a comment? I have no clue what the third argument is